### ISSUE TITLE
Add a public testing module

### ISSUE THREAD

**Issue by gabrieldemarmiesse** [tensorflow/addons#1294]
**Describe the feature and the current behavior/state.**

Currently, tensorflow offers some low level functions and classes for testing: https://www.tensorflow.org/api_docs/python/tf/test 

But overall, we show in the test suite that we lack tests that we could apply to any keras objects as seen in #992

My proposal is to take #992 and that we expose publicly some of the functions created. I can be a codeowner. Some good candidates would be convergence tests for optimizers, a copy of `layer_test` for layers and serialization tests for all keras objects.

We could then use those functions in our test suite and other projects could use them too.

**Relevant information**
- Are you willing to contribute it (yes/no): yes
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where): no
- Is there already an implementation in another framework? (if so, where): keras-team/keras tf.keras
- Was it part of tf.contrib? (if so, where): no

**Which API type would this fall under (layer, metric, optimizer, etc.)**

`tfa.testing`

**Who will benefit with this feature?**

Everyone who make new keras objects and want to test them.

**Any other info.**

#992

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
